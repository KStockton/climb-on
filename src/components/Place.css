.module {
  border: 2px solid #65655E;
  border-radius: 5px;
  box-shadow: 0 0 1px lightgrey;
}

.Place {
  margin: 15px auto;
  max-width: 750px;
  padding: 25px;
  text-align: left;
}

.Place > * {
  margin: 0;
}

.Place-header-container {
  display: flex;
  flex-direction: row;
  justify-content: space-between;
  align-items: center;
  margin-bottom: 15px;
}

.Place-info {
  flex: 3;
  height: 150px;
  display: flex;
  flex-direction: column;
  justify-content: flex-start;
}

.Place-name {
  font-size: 35px;
  margin: 0;
  font-weight: 500;
}

.Place-nearest-town {
  margin: 0;
  font-weight: 500;
}

.Place-photo {
  flex: 2;
  border: 1px solid black;
  height: 150px;
}

.Place-routes-label {
  margin: 0;
  font-weight: 500;
  font-size: 25px;
}

.Place-divider {
  border-style: none;
  border-bottom: 1px solid #d8d9db;
  margin: 5px 0 10px 0;
}

.Place-routes-container {
  border: 1px solid #65655E;
  box-shadow: 0 0 2px lightgrey inset;
  padding: 10px;
  border-radius: 3px;
  margin-top: 10px;
  display: flex;
  flex-direction: row;
  overflow-x: auto;
}