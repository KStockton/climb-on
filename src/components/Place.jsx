import React, { Component } from 'react';
import Route from './Route.jsx';

import "./Place.css";

class Place extends Component {

  render() {
    const {area, closestTown, routes, photo} = this.props;
    return (
      <article className="Place module">
        <section className="Place-header-container">
          <div className="Place-info">
            <h3 className="Place-name">Golden Cliffs</h3>
            {/* <h3 className="Place-name">{area}</h3> */}
            <h4 className="Place-nearest-town">Closest City: Golden</h4>
            {/* <h4 className="Place-nearest-town">{closestTown}</h4> */}
          </div>
          <img className="Place-photo" src={photo} alt="location photo"/>
        </section>
        <h3 className="Place-routes-label">Routes</h3>
        {/* <hr className="Place-divider"/> */}
        <section className="Place-routes-container">
          {/* {
            routes.map(route =>{
              return <Route routeName={route.routeName}/>
            })
          } */}
            <Route routeName="Louise" area="Brown Cloud Crags" difficultyLevel="5.8" type={["sport"]}/>
            <Route routeName="Louise" area="Brown Cloud Crags" difficultyLevel="5.8" type={["sport"]}/>
            <Route routeName="Louise" area="Brown Cloud Crags" difficultyLevel="5.8" type={["sport"]}/>
            <Route routeName="Thelma" area="Brown Cloud Crags" difficultyLevel="5.7" type={["sport"]}/>
            <Route routeName="Mister Squirrel Places a Nut" area="Overhang Area" difficultyLevel="5.11c" type={["sport", "top-rope"]}/>
        </section>
      </article>
    )
  }
}

export default Place
