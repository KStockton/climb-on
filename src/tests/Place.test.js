import React from 'react';
import Place from '../components/Place';
import { shallow } from 'enzyme';

const mockPlace = { 
  "place": "Golden Cliffs",
  "closestTown": "Golden, Colorado",
  "climbingId": 1,
  "routes": [
    { "climbingPlaceId": 1,
      "area": "Brown Cloud Crags",
      "routeName": "Louise",
      "difficultyLevel": "5.8",
      "type": ["sport"] },
    { "climbingPlaceId": 1,
      "area": "Brown Cloud Crags",
      "routeName": "Thelma",
      "difficultyLevel": "5.7",
      "type": ["sport"] }
  ]
}

describe('SearchForm', () => {
  let wrapper;

  beforeEach(() => {
    wrapper = shallow(
      <Place
        key={0}
        place={mockPlace.place}
        closestTown={mockPlace.closestTown}
        climbingId={mockPlace.climbingId}
        routes={mockPlace.routes}
      />
    )
  });

  it('should match the snapshot with all the data passed in', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('should display No routes to display if there are no routes', () => {
    let wrapper2 = shallow(
      <Place
        key={0}
        place={mockPlace.place}
        closestTown={mockPlace.closestTown}
        climbingId={mockPlace.climbingId}
        routes={[]}
      />
    )

    expect(wrapper2).toMatchSnapshot();
  });









});