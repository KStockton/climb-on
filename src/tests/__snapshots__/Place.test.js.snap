// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchForm should display No routes to display if there are no routes 1`] = `
<article
  className="Place module"
>
  <section
    className="Place-header-container"
  >
    <div
      className="Place-info"
    >
      <h3
        className="Place-name"
      >
        Golden Cliffs
      </h3>
      <h4
        className="Place-nearest-town"
      >
        Closest City: 
        Golden, Colorado
      </h4>
    </div>
  </section>
  <section
    className="Place-routes-container"
  >
    <h3
      className="Place-no-results-text"
    >
      No routes to display
    </h3>
  </section>
</article>
`;

exports[`SearchForm should match the snapshot with all the data passed in 1`] = `
<article
  className="Place module"
>
  <section
    className="Place-header-container"
  >
    <div
      className="Place-info"
    >
      <h3
        className="Place-name"
      >
        Golden Cliffs
      </h3>
      <h4
        className="Place-nearest-town"
      >
        Closest City: 
        Golden, Colorado
      </h4>
    </div>
  </section>
  <section
    className="Place-routes-container"
  >
    <Route
      area="Brown Cloud Crags"
      climbingPlaceId={1}
      difficultyLevel="5.8"
      key="0"
      routeName="Louise"
      type={
        Array [
          "sport",
        ]
      }
    />
    <Route
      area="Brown Cloud Crags"
      climbingPlaceId={1}
      difficultyLevel="5.7"
      key="1"
      routeName="Thelma"
      type={
        Array [
          "sport",
        ]
      }
    />
  </section>
</article>
`;
